[wailydest] iP

Wailydest opened the pull request:
# A brand new bot capable of memorizing your entire life.

* A true sigma
* Has sigma memory
* Dominates your file system for memorization purposes
* Has the _cleanest_ syntax in the world :sunglasses::
```java
        if (input.equals("bye")) return new ExitCommand();
        if (input.equals("list")) return new ListCommand();
        if (input.startsWith("mark") || input.startsWith("unmark")) return new MarkingCommand(input);
        if (input.startsWith("delete")) return new DeleteCommand(input);
        if (input.startsWith("todo") || input.startsWith("deadline") || input.startsWith("event")) {
            return new AddTaskCommand(input);
        }
        if (input.startsWith("find")) return new FindCommand(input);
        throw new UnknownCommandInputException(input);
```

Latest feature updates:
1. Got style (follows code style standards)
2. Got _some_ documentation
3. Can help you find your memories if you're completely lost
4. Got GHMD in the [PR description](https://github.com/nus-cs2103-AY2425S2/ip/pull/285) (you're reading it now)

**Public poll**: Do you like sigma bot?
- [x] Yes
- [x] More than

> "I tried to fix an Error in my Java code. Now I've got ErrorFactory" - some developer

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Wailydest committed:
change bot's name

Wailydest committed:
add echo feature

Wailydest committed:
add list

Wailydest committed:
add mark as done feature

Wailydest committed:
add task types

Wailydest committed:
add automated testing

Wailydest committed:
added error handling

Wailydest committed:
add delete feature

Wailydest committed:
add save feature

Wailydest committed:
Merge branch 'branch-Level-7'

Wailydest committed:
handle corrupt data

Wailydest committed:
add time formatting

Wailydest committed:
Merge branch 'branch-Level-8'

Wailydest committed:
add more oop

Wailydest committed:
Merge branch 'branch-A-MoreOOP'

Wailydest committed:
Merge branch 'add-gradle-support'

Wailydest committed:
integrate gradle into project

Wailydest committed:
add junit tests

AndreChia commented on src/main/java/sigmabot/ui/commands/MarkingCommand.java:
```diff
@@ -0,0 +1,41 @@
+
+public final class MarkingCommand extends Command {
+    final private int taskNumber;
+    final private boolean toMark;
```
Would isMarkingTask be a more descriptive name than toMark?

AndreChia commented on src/main/java/sigmabot/tasks/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+        super(t);
+        this.by = t.by;
+    }
+    @Override
```
Would cloneTask() be a better method name than copy()?

AndreChia commented on src/main/java/sigmabot/ui/commands/AddTaskCommand.java:
```diff
@@ -0,0 +1,47 @@
+        if (!matcher.find()) throw new IncorrectTaskFormat(input);
+        String description = matcher.group(1).trim();
+        if (input.startsWith("deadline")) {
+            var matcherBy = Pattern.compile("/by([^/]*)").matcher(input);
```
Would it be better to rename matcherBy to deadlineMatcher? matcherBy is unclear and it doesn’t indicate that it’s matching a deadline.

AndreChia replied:
Would it be better to rename matcherBy to deadlineMatcher? matcherBy is unclear and it doesn’t indicate that it’s matching a deadline.

AndreChia commented on src/main/java/sigmabot/tasks/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+import java.time.LocalDateTime;
+
+public final class Deadline extends Task {
+    LocalDateTime by;
```
Would dueDate or deadline be a better name than 'by'? 'by' is too short and ambiguous—it doesn’t clearly indicate that it refers to a due date.

AndreChia reviewed (commented):
Overall, LGTM! Keep up the excellent work! 👍

AndreChia commented on src/main/java/sigmabot/tasks/Deadline.java:
```diff
@@ -0,0 +1,49 @@
+        this.by = t.by;
+    }
+    @Override
+    protected Task copy() {
```
Would cloneTask() be a better method name than copy()?

The word "copy" is too general—it’s unclear if it means duplicating the task object or just copying some properties.

Wailydest replied:
In the context of immutable classes, I though copy is a universally used method name, referencing a complete duplication

Wailydest committed:
add javadoc

Wailydest committed:
format the code

Wailydest committed:
add find feature

Wailydest committed:
Merge branch 'branch-A-JavaDoc'

Wailydest committed:
resolve merge conflicts

Wailydest committed:
merge with branch-Level-9

Wailydest committed:
adjust namings

jamesberyl commented on src/main/java/sigmabot/exception/IncorrectDeleteFormat.java:
```diff
@@ -0,0 +1,14 @@
+/**
+ * An exception thrown when the user inputs an incorrect delete format.
+ */
+public class IncorrectDeleteFormat extends SigmabotInputException {
```
Should this class name follow the convention of ending with Exception? 

It might be clearer as IncorrectDeleteFormatException, making it explicit that this is an exception class. I noticed the same in a few other places too.

cscms03 replied:
I think the name should end with Exception, following the naming convention. Great use of custom exception classes though.

jamesberyl commented on src/main/java/sigmabot/ui/Ui.java:
```diff
@@ -0,0 +1,64 @@
+package sigmabot.ui;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
+import sigmabot.exception.SigmabotException;
+import sigmabot.exception.SigmabotInputException;
+import sigmabot.ui.commands.Command;
+import sigmabot.ui.commands.ExitCommand;
```
I like how you kept to a consistent ordering of import statements, following the Java standard convention.

jamesberyl commented on src/main/java/sigmabot/ui/commands/MarkingCommand.java:
```diff
@@ -0,0 +1,55 @@
+ */
+public final class MarkingCommand extends Command {
+    final private int taskNumber;
+    final private boolean isMarkingTask;
```
Good use of a boolean naming convention.

jamesberyl commented on src/main/java/sigmabot/ui/commands/Command.java:
```diff
@@ -0,0 +1,38 @@
+        if (input.startsWith("todo") || input.startsWith("deadline") || input.startsWith("event")) {
+            return new AddTaskCommand(input);
+        }
+        if (input.startsWith("find")) return new FindCommand(input);
```
You could consider using a switch statement here to improve readability.

jamesberyl commented on src/main/java/sigmabot/ui/Sigmabot.java:
```diff
@@ -0,0 +1,32 @@
+    private final static String DATA_DIR_NAME = "data";
+    /**
+     * The name of the data file inside the data directory.
+     */
```
You could consider using private static final, which is the conventional order in Java instead of private final static.

jamesberyl reviewed (commented):
Overall, LGTM. The code was easy to read and mostly follows Java coding standards. There was also a good use of OOP principles, breaking down larger classes into smaller ones. Good job! I especially liked the use of immutability and exception handling, which makes the code more robust and maintainable.

cscms03 commented on src/main/java/sigmabot/exception/UnknownCommandInputException.java:
```diff
@@ -0,0 +1,17 @@
+package sigmabot.exception;
+
+/**
```
great effort for making very very specific exception classes for error handling!

cscms03 commented on src/main/java/sigmabot/tasks/Deadline.java:
```diff
@@ -0,0 +1,65 @@
+
+    }
+
+    private Deadline(Deadline t) {
```
t as variable name could be a bit confusing to read at first! maybe make name more specific?

cscms03 commented on src/main/java/sigmabot/tasks/Storage.java:
```diff
@@ -0,0 +1,100 @@
+     * @return an ArrayList list of the tasks loaded from the data file.
+     * @throws SigmabotDataException if the data file cannot be read.
+     */
+    public ArrayList<Task> load() throws SigmabotDataException {
```
more specific naming "loadTask()" could be better? just suggesting another option!

cscms03 reviewed (commented):
Overall, naming conventions can be more specific to strictly comply with coding standard. I find the code was neat and easy to read through!

Wailydest committed:
add GUI

Wailydest committed:
Merge branch 'branch-Level-10'

Wailydest committed:
add assertion

There's a double check for accordance of the task type.

One of the checks must not be executed.

Perform assertion within the AddTaskCommand class, as it's guaranteed
that the task already has an appropriate type.

Assertion is used as reaching this line clearly indicates inconsistency
in code and in logic.

Wailydest committed:
clean up code

Wailydest committed:
Merge pull request #2 from Wailydest/branch-A-Assertions

add assertion

Wailydest committed:
update header comments

Wailydest committed:
Merge branch 'master' into branch-A-CodeQuality

Wailydest committed:
Merge pull request #3 from Wailydest/branch-A-CodeQuality

Code quality improvements

Wailydest committed:
add tag feature

Wailydest committed:
update javadoc

Wailydest committed:
display application name

Wailydest committed:
add image

Wailydest committed:
add user guide

Wailydest committed:
update user guide

Wailydest committed:
decrease image size

Wailydest committed:
Update README.md

Wailydest committed:
Update README.md
